src/api/spec/controllers/webui/webui_controller_spec.rb

```diff
@@ -135,4 +135,168 @@ def edit
       end
     end
   end
+
+  describe '#current_announcement' do
```

Comment:
This should be a unit test of `StatusMessage.visible_for_current_user?` instead

Reply:
Both `current_announcement` and `visible_for_current_user?`disappear in a more recent commit. We are working on squashing commits right now. I'll let you know when it is ready to review again.